picard/tagger.py

```diff
@@ -1246,12 +1246,58 @@ def sighandler(self):
         self.signalnotifier.setEnabled(True)
 
 
-def version():
-    print("%s %s %s" % (PICARD_ORG_NAME, PICARD_APP_NAME, PICARD_FANCY_VERSION_STR))
+class PicardArgs:
 
+    def __init__(self, argparse_args=None):
+        if argparse_args is not None:
+            self.load_from_argparse(argparse_args)
 
-def longversion():
-    print(versions.as_string())
+    def load_from_argparse(self, argparse_args):
+        self.CONFIG_FILE = argparse_args.config_file
+        self.DEBUG = argparse_args.debug
+        self._EXEC = argparse_args.exec
+        self.NO_PLAYER = argparse_args.no_player
+        self.NO_RESTORE = argparse_args.no_restore
+        self.NO_PLUGINS = argparse_args.no_plugins
+        self.NO_CRASH_DIALOG = argparse_args.no_crash_dialog
+        self.STAND_ALONE_INSTANCE = argparse_args.stand_alone_instance
+        self.VERSION = argparse_args.version
+        self.LONG_VERSION = argparse_args.long_version
+        self._FILE_OR_URL = argparse_args.FILE_OR_URL
+        self.PROCESSABLE = []
+
+        self.__parse_loadable_items()
+
+    def __parse_loadable_items(self):
+        for x in self._FILE_OR_URL:
+            if not urlparse(x).netloc:
+                x = os.path.abspath(x)
+            self.PROCESSABLE.append(x)
+
+        if self._EXEC:
+            for e in self._EXEC:
+                if "HELP" in [x.upper().strip() for x in e]:
+                    print_help_for_commands()
+                    sys.exit(0)
+                args = e[1:] if len(e) > 1 else []
+                for arg in args:
+                    self.PROCESSABLE.append(f"command://{e[0]} {arg}")
```

Comment:
By defaulting to an empty list rather than `['']` I think that any commands without arguments such as `-e CLUSTER_ALL` will not be added to `self.PROCESSABLE`.  That's why my suggestion was to default to `['']`.